### PR TITLE
No dup recv complete call in passive recv

### PR DESCRIPTION
**Pull request by qzhuyan**
fix: #132

### CODE REVIEW

File: docs/internals.org [hunk 1]
```diff
@@ -409,3 +409,89 @@ In quicer, for best practice, you could use quicer_* modules to manage the liste
...
+If `N` is once, the stream recv mode will become passive mode and no `{quic, passive, stream_handle(), undefined}`
+will be delivered to the owner.
+
+When the stream enters passive mode the MsQuic receiving callback is disabled.
+
+*** Callback handle `QUIC_STREAM_EVENT_RECEIVE` in Passive mode
+
+Quicer utilize the "Asynchronous receive" and "Partial Data Acceptance" functionalities provided by MsQuic.
+
+In passive mode, stream *recv callback* returns `QUIC_STATUS_PENDING` to enable "Asynchronous receive", this pasues the further
```
Comment (thalesmg) [inline docs/internals.org:460]: Suggestion:

```suggestion
In passive mode, stream *recv callback* returns `QUIC_STATUS_PENDING` to enable "Asynchronous receive", this pauses the further
```

File: docs/internals.org [hunk 2]
```diff
@@ -409,3 +409,89 @@ In quicer, for best practice, you could use quicer_* modules to manage the liste
...
+quicer:setopts(stream(), active, Mode).
+#+end_src
+
+The active *Mode* is compatible with inet:setopts/2 as in https://www.erlang.org/doc/man/inet.html#setopts-2
+
+** Implementations
+
+*** Callback handle `QUIC_STREAM_EVENT_RECEIVE` with received binary in Active mode
+
+The whole recv bytes will be delivered to the owner process. The integer `N` in {active, N} will be decreased 1.
```
Comment (thalesmg) [inline docs/internals.org:447]: Suggestion:

```suggestion
The whole recv bytes will be delivered to the owner process. The integer `N` in {active, N} will be decreased by 1.
```

File: docs/internals.org [hunk 3]
```diff
@@ -409,3 +409,89 @@ In quicer, for best practice, you could use quicer_* modules to manage the liste
...
+*** Callback handle `QUIC_STREAM_EVENT_RECEIVE` in Passive mode
+
+Quicer utilize the "Asynchronous receive" and "Partial Data Acceptance" functionalities provided by MsQuic.
+
+In passive mode, stream *recv callback* returns `QUIC_STATUS_PENDING` to enable "Asynchronous receive", this pasues the further
+receiving callbacks until `quicer:recv` is called. The `is_recv_pending` of stream ctx is set to mark the recv is pending and wait for
+a `StreamReceiveComplete` to complete the receive.
+
+If `quicer:recv` is called before recv callback (is_recv_pending is not set) which means the application
+is blocked form receiving the data. Quicer will send a 'recv continue' msg as below to the stream owner to unblock the owner.
```
Comment (thalesmg) [inline docs/internals.org:465]:
```suggestion
is blocked from receiving the data. Quicer will send a 'recv continue' msg as below to the stream owner to unblock the owner.
```

File: docs/internals.org [hunk 4]
```diff
@@ -409,3 +409,89 @@ In quicer, for best practice, you could use quicer_* modules to manage the liste
...
+The side effect of calling `StreamReceiveComplete` is it completes the *async recv* handling and reenables the recv callback if all
+data in the recv buffer is consumed. If it is partially consumed, the stream receving will be left disabled as it is in passive mode.
+NOTE, duplicate `StreamReceiveComplete` calls could cause undefined behaviors. see [MsQuic/docs/Streams.md]
+
+If FALSE, NIF will mark `is_wait_for_data` in stream ctx and return `{error, not_ready}` to the caller and the caller should
+wait for recv `continue` message mentioned above.
+
+*** quicer:recv/2 call in Active Mode
+
+Should return error.
```
Comment (thalesmg) [inline docs/internals.org:484]: Tiny nit: maybe add a test case for this?

**Review by thalesmg — APPROVED** [review]

**Review by thalesmg — APPROVED** [review]

File: docs/internals.org (hunk 4 above)
Comment (qzhuyan) [reply docs/internals.org:484]: Yes we have already.
https://github.com/qzhuyan/quic/blob/fix/william/no-dup-stream-complete-call-in-passive-recv/test/quicer_SUITE.erl#L903

**Pull request merged**